java-checks/src/test/files/checks/PrintfMisuseCheck.java

```diff
@@ -182,13 +185,18 @@ void foo(Calendar c){
     slf4jLog.debug("message {} {}", new Object[]{1, 2, 3}); // Noncompliant
     slf4jLog.debug("message {} {} {}", new Object[]{1, 2, 3}); // compliant
     slf4jLog.debug("message ", new Exception());
-    slf4jLog.debug("message {}", new Exception());
+    slf4jLog.debug("message {}", new Exception()); // Noncompliant {{Missing argument for the first parameter.}}
 
-    slf4jLog.error("message {}");
+    slf4jLog.error("message {}"); // Noncompliant {{Missing argument for the first parameter.}}
+    slf4jLog.error("message {}", new Exception()); // Noncompliant {{Missing argument for the first parameter.}}
```

Comment:
to me the error message is wrong here. It should be something like `Remove this argument formatter because it won't be consumed`.